src/OrchardCore.Modules/OrchardCore.Title/Models/TitlePartSettings.cs

```diff
@@ -0,0 +1,15 @@
+namespace OrchardCore.Title.Models
+{
+    public class TitlePartSettings
+    {
+        /// <summary>
+        /// Gets or sets whether a user can define a custom title
+        /// </summary>
+        public bool? AllowCustomTitle { get; set; }
+
```

Comment:
@jptissot Looking great!

Here's how to set a default value of True in a setting.

See the media field, it has a default value attribute set to DefaultValue(true) for the Multiple field. So you should be able to use the same if you want

https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Media/Settings/MediaFieldSettings.cs

        [DefaultValue(true)]
        public bool Multiple { get; set; } = true;
So the attribute make json serialize it when set to false (because false is not the default value) but when it is constructed for the first time with no value, the = true will also set it.

Also you should not have to use a nullable `bool?`, as any call to `GetSettings` will create the `TitlePartSettings` object, even if it doesn't exist previously (so no migration required either), and will then set to whatever the default value is.

Reply:
I am pretty sure I tried this and was having issues.
Repro:
- Create Type with TitlePart
- Go into TitlePart settings, set it to false (uncheck the box)
- Save, The value is set properly in the driver
- Go bak to TitlePage settings page, box is checked when it should not.

I'll try again when I get a chance.

Reply:
Oh, you need both the attribute and the setting. Maybe that was what I was missing.

Reply:
Yes, that's it. The attribute tells Json how to handle the default value, but you still need to actually assign what you want as well
https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.defaultvalueattribute?view=netcore-3.0

I was going to say we could add a comment to
https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Metadata/Builders/ContentBuilderSettings.cs

But actually it's already there :)